models/Category.php

```diff
@@ -256,7 +256,7 @@ public static function resolveMenuItem($item, $url, $theme)
                 'items' => []
             ];
 
-            $categories = self::orderBy('name')->get();
+            $categories = self::has('posts')->orderBy('name')->get();
```

Comment:
is has short for whereHas?

Reply:
Yes. The category model already applies the 'published' scope to it's 'posts' relation.